examples/chef/common/chef-rvc-mode-delegate.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/util/af.h>
+#include <app/util/config.h>
+#include <cstring>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace RvcRunMode {
+
+const uint8_t ModeIdle     = 0;
```

Comment:
I am a little rusty, just coming back from leave. From what I gather, this is application specific modes and these mode values are not in the spec; how does a matter controller know about these values, so that they can send the values? Would they have to query using `GetModeLabelByIndex` first?

Reply:
@tehampson The mode ids are not in the spec. The application code is free to use whatever values it wants for them. However, the app code should use mode tags (which are in the spec) to help the clients understand the semantic of each mode.

Reply:
NVM, found it shortly afterward. It is queried using `EncodeSupportedModes`

Reply:
@plauric Thanks for the explanation! That game me some other context that I needed to look into